paddle/fluid/framework/data_layout_transform.cc

```diff
@@ -194,6 +194,9 @@ void innerTransDataLayoutFromMKLDNN(DataLayout in_layout, DataLayout out_layout,
         handler.AcquireReorder(reorder_dst_memory_p, reorder_src_memory_p);
 
     mkldnn::stream astream(cpu_engine);
+    #ifdef PADDLE_WITH_MKLDNN
```

Comment:
I think we do not need this guard. We are already in a section executed with MKLDNN.